src/lib.rs

```diff
@@ -1,3 +1,4 @@
 pub mod buffer_pool_manager;
 pub mod query_plan;
 pub mod shared;
+pub mod storage;
```

Comment:
Slight nitpick, could we call this `storage_handler` to line up with the plan?

Reply:
I was looking through every other database directory structure, and storage is the name of the folder, so I think it's good to be consistent w them - buffer pool manager should be moved into storage